x-pack/plugins/triggers_actions_ui/public/application/hooks/use_load_rule_aggregations.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { useState, useCallback } from 'react';
+import { RuleExecutionStatusValues } from '@kbn/alerting-plugin/common';
+import { loadRuleAggregations, LoadRuleAggregationsProps } from '../lib/rule_api';
+import { useKibana } from '../../common/lib/kibana';
+
+type UseLoadRuleAggregationsProps = Omit<LoadRuleAggregationsProps, 'http'> & {
+  onError: (message: string) => void;
+};
+
+export function useLoadRuleAggregations({
+  searchText,
+  typesFilter,
+  actionTypesFilter,
+  ruleExecutionStatusesFilter,
+  ruleStatusesFilter,
+  tagsFilter,
+  onError,
+}: UseLoadRuleAggregationsProps) {
+  const { http } = useKibana().services;
+
+  const [rulesStatusesTotal, setRulesStatusesTotal] = useState<Record<string, number>>(
+    RuleExecutionStatusValues.reduce(
```

Comment:
nit -> I think to avoid the cast
```suggestion
    RuleExecutionStatusValues.reduce<Record<string, number>>(
```